src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -291,15 +291,26 @@ protected function stream(string $view, array $parameters = [], StreamedResponse
     }
 
     /**
-     * Renders a form.
+     * Handles a form.
      *
-     * The FormView instance is passed to the template in a variable named "form".
-     * If the form is invalid, a 422 status code is returned.
+     * * if the form is not submitted, $render is called
+     * * if the form is submitted but invalid, $render is called and a 422 HTTP status code is set if the current status hasn't been customized
+     * * if the form is submitted and valid, $onSuccess is called, usually this method saves the data and returns a 303 HTTP redirection
+     *
+     * @param callable(FormInterface): Response $onSuccess
+     * @param callable(FormInterface): Response $render
      */
-    public function renderForm(string $view, FormInterface $form, array $parameters = [], Response $response = null): Response
+    public function handleForm(Request $request, FormInterface $form, callable $onSuccess, callable $render): Response
```

Comment:
Would it make sense to swap $request and $form? I feel like a method that ends with "form" should accept the form as 1st argument.